File: project/.sdk/tm/lua/core/control.lua
-- ProjectName SDK control

local Control = {}
Control.__index = Control


function Control.new(opts)
  opts = opts or {}
  local self = setmetatable({}, Control)
  self.throw_err = opts.throw_err
  self.err = nil
  self.explain = opts.explain
  return self
end


return Control

File: ts/project/.sdk/tm/swift/Sources/ProjectNameSDK/core/Control.swift
// ProjectName SDK - per-call control block.

import Foundation

public final class Control {
  // `throw` is a Swift keyword; the pipeline reads `throwErr == false`.
  public var throwErr: Bool? = nil
  public var err: Error? = nil
  // Explain and paging carry loose data + typed products (stored via .nat).
  public var explain: VMap? = nil
  public var actor: String = ""
  public var paging: VMap? = nil

  // Outbound streaming marker: the entity `stream` method sets this to the
  // caller-supplied streamable payload so the request builder / transport can
  // stream it as the request body.
  public var streamOut: Value? = nil

  public init() {}
}
